superset/jinja_context.py

```diff
@@ -93,18 +94,16 @@ def filter_values(column: str, default: Optional[str] = None) -> List[str]:
     :return: returns a list of filter values
     """
     form_data = json.loads(request.form.get("form_data", "{}"))
+    utils.harmonize_query_filters(form_data)
+
     return_val = []
-    for filter_type in ["filters", "extra_filters"]:
-        if filter_type not in form_data:
-            continue
-
-        for f in form_data[filter_type]:
-            if f["col"] == column:
-                if isinstance(f["val"], list):
-                    for v in f["val"]:
-                        return_val.append(v)
-                else:
-                    return_val.append(f["val"])
+    for f in form_data.get("filters", {}):
```

Comment:
I was working on filters recently and I'm pretty sure these are now under `adhoc_filters`

Reply:
This wan't completely familiar territory to me, so I might be wrong here. But currently `process_query_filters()` in `viz..py`, which is the first thing `query_obj()` calls, goes through a long step of transformations that first changes everything into `adhoc_filters`, and finally changes those to regular `filters`. I'm unsure why this happens, but my takeaway was that if `process_query_filters` is run, you are left with nothing but `filters`, despite how they started out.

Reply:
Oh wow. This is confusing. Looks like you looked into it deeper than I have. 👍

Reply:
Probably something for the refactor backlog. I took another pass at how this works in practice, and these are my takeaways:
- `connectors/sqla/model.py` has no logic for handling `adhoc_metrics`, so all filters are expected to be in base filter form: https://github.com/apache/incubator-superset/blob/master/superset/connectors/sqla/models.py#L837-L877 . The same goes for the legacy Druid connector.
- The new QueryObject used by the new Query API only supports base filters: https://github.com/apache/incubator-superset/blob/master/superset/common/query_object.py#L115 . I think we need to change this so that `adhoc_filters` are allowed in the constructor signature and are converted to base filters in `to_dict()`.

Anyway, in the short term, I think this fix is the right solution.